javalib/src/main/scala/java/util/NullRejectingHashMap.scala

```diff
@@ -80,6 +89,37 @@ private[util] class NullRejectingHashMap[K, V](
     super.remove(key)
   }
 
+  override def remove(key: Any, value: Any): Boolean = {
+    val old = this.get(key) // throws if `key` is null
+    if (old != null && old.equals(value)) { // false if `value` is null
+      super.remove(key)
+      true
+    } else {
+      false
+    }
+  }
+
+  override def replace(key: K, oldValue: V, newValue: V): Boolean = {
+    if (key == null || oldValue == null || newValue == null)
+      throw new NullPointerException()
+    val old = super.get(key)
+    if (oldValue.equals(old)) { // false if `old` is null
+      super.put(key, newValue)
+      true
+    } else {
+      false
+    }
+  }
+
+  override def replace(key: K, value: V): V = {
+    if (key == null || value == null)
+      throw new NullPointerException()
+    val old = super.get(key)
+    if (old != null)
+      super.put(key, value)
+    old
+  }
+
```

Comment:
Some of the `null` handling is inconsistent here: Sometimes it is delegated to other methods (notably `get`) when possible, sometimes not. Have you considered simply moving the implementations that are currently in ConcurrentHashMap.

Reply:
CHM now simply delegets to NullRejectingHashMap

Reply:
Good. But the `null` handling is still inconsistent: Some methods delegate to `get` to check if they `key` is non-null, some don't.

For simplicity: Simply move the implementations from `ConcurrentHashMap`?

Reply:
Delegate null check to `(this.)get|` or other method as much s possible. https://github.com/scala-js/scala-js/pull/4367/commits/f12100cd44447c752fc3402e05438eab68e12ed8

Reply:
Now some of the comments are gone. Please just take the implementations in CHM.

Reply:
https://github.com/scala-js/scala-js/pull/4367/commits/fa361b53962fb1b4a686f8a8093a61c04aa42a3b